File: Java files/src/main/java/com/cts/entities/Category.java
package com.cts.entities;

public class Category {

	private int categoryId;
	private String categoryName;
	private int parentCategoryId;

	public Category() {
		categoryId = 0;
		categoryName = "";
		parentCategoryId = 0;
	}

	public int getCategoryId() {
		return categoryId;
	}

	public void setCategoryId(int categoryId) {
		this.categoryId = categoryId;
	}

	public String getCategoryName() {
		return categoryName;
	}

	public void setCategoryName(String categoryName) {
		this.categoryName = categoryName;
	}

	public int getParentCategoryId() {
		return parentCategoryId;
	}

	public void setParentCategoryId(int parentCategoryId) {
		this.parentCategoryId = parentCategoryId;
	}
}

File: Java files/src/main/webapp/resources/libs/models/userMenuModels/category.js
/*! CTS Category Model */

var Category = Backbone.Model.extend({ 
	
	defaults: {
		categoryId: '',
		categoryName: '',
		parentCategoryId: ''
	},
	
	check: function() {
		
		var errors = {};
		
		/*if (this.get('categoryId') == '') {
			errors['#categoryId'] = getValue('mandatory');
			errorsFound = true;
		};*/
		
		if (this.get('categoryName') == '') {
			errors['#ticketCategoryDropbox'] = getValue('mandatory');
		} else {
			errors['#ticketCategoryDropbox'] = '';
		};
		
		return errors;
	}
});